#!/usr/bin/env python

PACKAGE = 'controller_executive'

from dynamic_reconfigure.parameter_generator_catkin import ParameterGenerator, double_t, bool_t, str_t

gen = ParameterGenerator()

gen.add("min_lookahead", double_t, 0, "Minimum lookahead", 0.2, 0, 5.0)
gen.add("max_lookahead", double_t, 0, "Maximum lookahead", 2.0, 0, 5.0)
gen.add("closest_point_index_search", double_t, 0, "Search for closest point index within this range of previous closest point", 2.0, 1.0, 20.0)
gen.add("max_acceleration", double_t, 0, "Maximum acceleration", 0.5, 0.01, 2.5)
gen.add("max_vel", double_t, 0, "Maximum velocity", 0.2, 0.01, 2.5)
gen.add("min_vel", double_t, 0, "Minimum velocity", 0.1, 0.01, 2.5)
gen.add("max_omega", double_t, 0, "Maximum angular velocity", 0.5, 0.01, 2.5)
gen.add("max_turn_radius", double_t, 0, "Maximum turn radius", 0.2, 0.01, 2.5)
gen.add("min_turn_radius", double_t, 0, "Minimum turn radius", 0.1, 0.01, 2.5)
gen.add("max_radius_omega", double_t, 0, "max_omega_radius", 20.0, 1.0, 50.0)
gen.add("max_y_deviation", double_t, 0, "max_y_deviation", 0.05, 0.01, 2.0)
gen.add("max_path_deviation", double_t, 0, "max_path_dev", 1.0, 0.0, 2.0)
gen.add("max_theta_deviation", double_t, 0, "max_theta_dev", 0.5, 0.0, 2.0)
gen.add("transform_tolerance", double_t, 0, "transform_tolerance", 0.5, 0.0, 5.0)
gen.add("holonomic_robot", bool_t, 0,
        "Set this to true if the robot being controlled can take y velocities and false otherwise", True)
gen.add("drive_type", str_t, 0, "Set this to differential/skid_steer/omni", "differential")
gen.add("on_the_spot_turn_allowed", bool_t, 0, "on_the_spot_turn_allowed", True)
gen.add("restore_defaults", bool_t, 0, "Retore to the default configuration", False)

exit(gen.generate(PACKAGE, "controller_executive_node", "PurePursuitController"))
